### PR TITLE
Building wifi hal code with WIFI_HAL_VERSION_3

### PR DESCRIPTION
**Pull request by KaviyaKumaresan**
REFPLTB-1457: Building wifi hal code with WIFI_HAL_VERSION_3

Reason for change: wifi_hal3.0 compilation for Turris-Omnia
Test Procedure: Build and test
Risk: low

Signed-off-by: KaviyaKumaresan <Kaviya.Kumaresan@ltts.com>

### CODE REVIEW

File: recipes-ccsp/ccsp/ccsp-wifi-agent.bbappend [hunk 1]
```diff
@@ -5,7 +5,7 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 LDFLAGS += " \
 	-lutopiautil \
 	   "
-CFLAGS_append = " -Wno-error"
+CFLAGS_append = " -Wno-error -DWIFI_HAL_VERSION_3"
```
Comment (mbajorski) [inline recipes-ccsp/ccsp/ccsp-wifi-agent.bbappend:8]: This is not needed as -DWIFI_HAL_VERSION_3 is only for hal-wifi-cfg80211 implementation.
Wifi hal headers do not contain WIFI_HAL_VERSION_3 flag

File: recipes-ccsp/ccsp/harvester_git.bbappend [hunk 2]
```diff
@@ -0,0 +1 @@
+CFLAGS_append  = " -DWIFI_HAL_VERSION_3"
```
Comment (mbajorski) [inline recipes-ccsp/ccsp/harvester_git.bbappend:1]: This is not needed as -DWIFI_HAL_VERSION_3 is only for hal-wifi-cfg80211 implementation.
Wifi hal headers do not contain WIFI_HAL_VERSION_3 flag

File: recipes-ccsp/hal/halinterface.bbappend [hunk 3]
```diff
@@ -10,3 +10,5 @@ do_turris_patches() {
         fi
 }
 addtask turris_patches after do_unpack before do_compile
+
```
Comment (mbajorski) [inline recipes-ccsp/hal/halinterface.bbappend:13]: This is not needed as -DWIFI_HAL_VERSION_3 is only for hal-wifi-cfg80211 implementation.
Wifi hal headers do not contain WIFI_HAL_VERSION_3 flag

**Review by mbajorski — CHANGES_REQUESTED** [review]
-DWIFI_HAL_VERSION_3 is only for hal-wifi-cfg80211 implementation.
Wifi hal headers do not contain WIFI_HAL_VERSION_3 flag

**Review by mbajorski — APPROVED** [review]
My mistake, This definition exist in hal interface

**Review by manigandanj — APPROVED** [review]

**Review by manigandanj — APPROVED** [review]

**Pull request merged**